### PR TITLE
fix: update Prismic terms to use the latest language style

### PR DESCRIPTION
**Pull request by angeloashmore**
## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR updates documentation and APIs to use the latest field name styles.

The following changes are included:

- "Integration Fields" → "integration" (e.g. "An integration field")
- "Custom Type" → "custom type" (e.g. "A custom type model")
- "Shared Slice" → "shared Slice" (e.g. "A shared Slice model")
- Lowercase all field names (e.g. "Rich Text" → "rich text")

### Integration fields

This package exports APIs that use the terms `IntegrationFields` and `integrationFields`. This PR deprecates, not removes, those APIs in favor of the renamed versions.

Please see the changed files for a full list of deprecated APIs.

cc @samlfair

## Checklist:

- [x] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

🦮

### CODE REVIEW

**Comment by lihbr** [comment]
Will have a look at it :)

Just letting you know, you can change the target branch of a PR after it's opened if need be~

File: src/createClient.ts [hunk 1]
```diff
@@ -234,7 +234,7 @@ export type ClientConfig = {
 	ref?: RefStringOrThunk;
 
 	/**
-	 * A list of Route Resolver objects that define how a document's `url` field
+	 * A list of Route Resolver objects that define how a document's `url` property
```
Comment (samlfair) [inline src/createClient.ts:237]: Nice catch 👍

File: src/types/model/customType.ts [hunk 2]
```diff
@@ -1,54 +1,54 @@
...
 > {
 	/**
-	 * The ID of the Custom Type model.
+	 * The ID of the custom type model.
 	 */
 	id: ID;
 
 	/**
-	 * The human readable name of the Custom Type Model.
+	 * The human readable name of the custom type Model.
```
Comment (samlfair) [inline src/types/model/customType.ts:19]: Might want to lowercase "Model" for consistency.

File: src/types/model/link.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export interface CustomTypeModelLinkField {
 }
 
 /**
- * Type of a Link Custom Type field.
+ * Type of a Link custom type field.
```
Comment (samlfair) [inline src/types/model/link.ts:22]: Should `Link` be lowercased here?

File: src/types/value/sharedSlice.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import type { SharedSliceVariation } from "./sharedSliceVariation";
 
 /**
- * Shared Slice
+ * A Shared Slice.
```
Comment (samlfair) [inline src/types/value/sharedSlice.ts:4]: Is it okay if we lowercase `Shared`? It didn't come up in my review of the documentation, but I don't think it needs to be title case.

File: src/types/value/sharedSliceVariation.ts [hunk 5]
```diff
@@ -1,5 +1,8 @@
 import type { AnyRegularField } from "./types";
 
+/**
+ * A Shared Slice variation.
```
Comment (samlfair) [inline src/types/value/sharedSliceVariation.ts:4]: Same comment about `Shared`.

File: src/types/value/slice.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import type { AnyRegularField } from "./types";
 
 /**
- * Slice - Sections of your website
+ * A Slice - sections of your website.
```
Comment (samlfair) [inline src/types/value/slice.ts:4]: @angeloashmore Could we change this to "sections of your webpages"?

To me, a section of a website is a group of pages. A section of a webpage is a group of elements.

File: src/types/model/customType.ts (hunk 2 above)
Comment (angeloashmore) [reply src/types/model/customType.ts:19]: Probably missed this one using find and replace. I checked the rest of the repo for "custom type Model" and confirmed there are no other instances of this. 👍

File: src/types/model/link.ts (hunk 3 above)
Comment (angeloashmore) [reply src/types/model/link.ts:22]: Yep! Thanks

File: src/types/value/sharedSlice.ts (hunk 4 above)
Comment (angeloashmore) [reply src/types/value/sharedSlice.ts:4]: Yes, let's do that. I updated all instances of "Shared Slice" with "shared Slice".

File: src/types/value/slice.ts (hunk 6 above)
Comment (angeloashmore) [reply src/types/value/slice.ts:4]: Done!

**Pull request merged**